### ISSUE TITLE
RealDebrid account 

### ISSUE THREAD

**Issue by asmoday83** [pyload/pyload#3476]
### Describe the problem

I'm trying to add the real-debrid.com account (username / apitoken) it fails with a timeout.
Sometimes the login succeeds but becomes invalid according to pyload's accounts page (Status = "not valid" and Premium = "no")

### Debug log

```
842 | 17.11.2019 15:26:35 | ERROR | ACCOUNT RealdebridCom: Could not login user `asm*******` \| (28, 'Connection timed out after 30001 milliseconds')
-- | -- | -- | --
841 | 17.11.2019 15:26:05 | INFO | ACCOUNT RealdebridCom: Relogin user `asm*******`...
840 | 17.11.2019 15:26:05 | INFO | ACCOUNT RealdebridCom: Adding user `asm*******`...
```

**Comment by GammaC0de** [comment]
Password for Real-debrid should be the API token,
Get it from: https://real-debrid.com/apitoken

Also, maybe they banned your IP,
Try putting the URL `https://api.real-debrid.com/rest/1.0/user?auth_token=<YOUR_API_KEY>` and see what response you get

**Comment by asmoday83** [comment]
I'm aware of that; after my first unsuccessful attempt with my password I found that information and how to obtain the token.

I don't think, that my IP is banned since I tried JDownloader and it works so far. Pyload still doesn't accept my credentials.
I tried the URL you gave me and got a JSON(?) reply with my userID, email and expiration date for my premium account; so I think that's fine, too.

**Comment by GammaC0de** [comment]
On what OS pyload is installed? Is it the same machine you have tried the URL with the browser?

**Comment by asmoday83** [comment]
OS is a headless dietpi. At first I tried it on my laptop, but now I also tried it with curl on the affected machine > same response with all my data in terminal.

**Comment by GammaC0de** [comment]
OK, so it should work then.

Is this a new pyload installation?

**Comment by asmoday83** [comment]
No, I wouldn't say that it's a new installation. I think I did it 1 to 2 years ago. But I regularly do a `git pull` on my installation directory (also yesterday).
Should I delete the .pyload directory to "reset" it?

**Comment by GammaC0de** [comment]
You can try deleting the userplugins directory and then restart pyload. 

 
 
  asmoday83<notifications@github.com> wrote:   
No, I wouldn't say that it's a new installation. I think I did it 1 to 2 years ago. But I regularly do a git pull on my installation directory (also yesterday).
Should I delete the .pyload directory to "reset" it?

**Comment by asmoday83** [comment]
Just tried it; same problem as before =/
```
18.11.2019 07:19:19 | ERROR | ACCOUNT RealdebridCom: Could not login user `asm*******` \| (28, 'Connection timed out after 30001 milliseconds')
-- | -- | -- | --
18.11.2019 07:18:49 | DEBUG | ACCOUNT RealdebridCom: LOAD URL https://api.real-debrid.com/rest/1.0/user \| redirect=True \| cookies=True \| get={'auth_token': '**********'} \| req=None \| decode=True \| multipart=False \| post={} \| ref=True \| just_header=False
18.11.2019 07:18:49 | INFO | ACCOUNT RealdebridCom: Relogin user `asm*******`...
18.11.2019 07:18:49 | DEBUG | ACCOUNT RealdebridCom: Reached login timeout for user `asm*******`
18.11.2019 07:18:49 | INFO | ACCOUNT RealdebridCom: Adding user `asm*******`...
```

**Comment by GammaC0de** [comment]
Does this happen right after you add the account to pyload?

Can you provide a debug log from the point pyload startup untill this error?

**Comment by asmoday83** [comment]
here's a log from starting pyload _without_ the account to the point after trying to add it:
```
75      19.11.2019 14:12:26     INFO    Starting pyLoad 0.4.9
76      19.11.2019 14:12:26     INFO    Using home directory: /home/pyload/.pyload
77      19.11.2019 14:12:26     DEBUG   Deleting old pidfile pyload.pid
78      19.11.2019 14:12:26     DEBUG   Remote activated: True
79      19.11.2019 14:12:28     DEBUG   created index of plugins
80      19.11.2019 14:12:28     INFO    Activated plugins: ClickNLoad, EventMapper, ExternalScripts, UnSkipOnFail, UpdateManager, UserAgentSwitcher, XFileSharing
81      19.11.2019 14:12:28     INFO    Deactivate plugins: AndroidPhoneNotify, AntiStandby, AntiVirus, BypassCaptcha, Captcha9Kw, Checksum, CloudFlareDdos, DeathByCaptcha, DeleteFinished, DiscordNotifier, DownloadScheduler, ExpertDecoders, ExtractArchive, HotFolder, IRC, ImageTyperz, JustPremium, LinkFilter, LinkdecrypterComHook, LogMarker, MergeFiles, MultiHome, PushBullet, PushOver, RestartFailed, SkipRev, TransmissionRPC, WindowsPhoneNotify, XMPP
82      19.11.2019 14:12:28     INFO    Downloadtime: True
83      19.11.2019 14:12:28     INFO    Starting ThriftBackend: 0.0.0.0:7227
84      19.11.2019 14:12:31     INFO    Starting builtin webserver: 0.0.0.0:8001
85      19.11.2019 14:12:31     INFO    Free space: 604.62 GiB
86      19.11.2019 14:12:31     INFO    Activating Accounts...
87      19.11.2019 14:12:31     INFO    Activating Plugins...
88      19.11.2019 14:12:31     DEBUG   ADDON UpdateManager: LOAD URL https://raw.githubusercontent.com/pyload/updates/master/plugins.txt | redirect=True | cookies=True | get={'v': '0.4.9'} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
89      19.11.2019 14:12:31     DEBUG   ADDON ExternalScripts: No script found under folder `pyload_start`
90      19.11.2019 14:12:32     INFO    ADDON XFileSharing: Auto-discover new hosters
91      19.11.2019 14:12:32     DEBUG   ADDON XFileSharing: Pattern for hosters: (?:https?://(?:www\.)?)(?!(?:www\.)?(?:180upload\.com|clicknupload\.org|cramit\.in|ddl\.to|easybytez\.com|exashare\.com|filejoker\.net|fileom\.com|filerio\.com|fistfast\.net|hugefiles\.net|junocloud\.me|katfile\.com|movreel\.com|nosupload\.com|novafile\.com|rapidfileshare\.net|rarefile\.net|safesharing\.eu|secureupload\.eu|streamcloud\.eu|suprafiles\.me|tusfiles\.net|uploadrocket\.net|vidplay\.net))(?P(?:[\d.]+|[\w\-^_]{3,63}(?:\.[a-zA-Z]{2,})+)(?:\:\d+)?)/(?:embed-)?\w{12}(?:\W|$)
92      19.11.2019 14:12:32     INFO    ADDON XFileSharing: Auto-discover new crypters
93      19.11.2019 14:12:32     DEBUG   ADDON XFileSharing: Pattern for crypters: (?:https?://(?:www\.)?)(?!(?:www\.)?(?:easybytez\.com|tusfiles\.net))(?P(?:[\d.]+|[\w\-^_]{3,63}(?:\.[a-zA-Z]{2,})+)(?:\:\d+)?)/(?:user|folder)s?/\w+
94      19.11.2019 14:12:32     INFO    pyLoad is up and running
95      19.11.2019 14:12:36     INFO    ADDON ClickNLoad: Proxy listening on 0.0.0.0:9666
96      19.11.2019 14:12:36     ERROR   ADDON ClickNLoad: [Errno 98] Address already in use
97      19.11.2019 14:12:37     INFO    ADDON UpdateManager: pyLoad is up to date!
98      19.11.2019 14:12:37     DEBUG   ADDON UpdateManager: LOAD URL https://raw.githubusercontent.com/pyload/updates/master/plugins.txt | redirect=True | cookies=True | get={'v': '0.4.9'} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
99      19.11.2019 14:12:37     DEBUG   ADDON UpdateManager: Requested deletion of plugins: [(u'accounts', u'BayfilesCom'), (u'accounts', u'BillionuploadsCom'), (u'accounts', u'BitshareCom'), (u'accounts', u'CyberlockerCh'), (u'accounts', u'DdlstorageCom'), (u'accounts', u'EgoFilesCom'), (u'accounts', u'File4safeCom'), (u'accounts', u'FileParadoxIn'), (u'accounts', u'FreeWayMe'), (u'accounts', u'HotfileCom'), (u'accounts', u'Keep2shareCc'), (u'accounts', u'LetitbitNet'), (u'accounts', u'LinestorageCom'), (u'accounts', u'LomafileCom'), (u'accounts', u'MegaDebrid'), (u'accounts', u'MultiDebridCom'), (u'accounts', u'NetloadIn'), (u'accounts', u'NowVideoAt'), (u'accounts', u'Premium4Me'), (u'accounts', u'RapidshareCom'), (u'accounts', u'ReloadCc'), (u'accounts', u'RyushareCom'), (u'accounts', u'ShareRapidCom'), (u'accounts', u'ShareonlineBiz'), (u'accounts', u'StahnuTo'), (u'accounts', u'UnrestrictLi'), (u'accounts', u'UploadableCh'), (u'accounts', u'UploadingCom'), (u'accounts', u'Vipleech4uCom'), (u'accounts', u'WarserverCz'), (u'accounts', u'XFileSharingPro'), (u'captcha', u'OCR'), (u'captcha', u'ShareonlineBiz'), (u'captcha', u'captcha'), (u'container', u'DLC_25'), (u'container', u'DLC_26'), (u'container', u'DLC_27'), (u'container', u'LinkList'), (u'crypter', u'C1neonCom'), (u'crypter', u'DailymotionBatch'), (u'crypter', u'DontKnowMe'), (u'crypter', u'DownloadVimeoCom'), (u'crypter', u'Go4Up'), (u'crypter', u'HotfileFolderCom'), (u'crypter', u'ImgurComAlbum'), (u'crypter', u'JunocloudMeFolder'), (u'crypter', u'Movie2kTo'), (u'crypter', u'OneKhDe'), (u'crypter', u'RapidfileshareNetFolder'), (u'crypter', u'ShareRapidComFolder'), (u'crypter', u'UploadableChFolder'), (u'crypter', u'XFileSharingProFolder'), (u'crypter', u'YoutubeBatch'), (u'hooks', u'AlldebridCom'), (u'hooks', u'AlldebridComHook'), (u'hooks', u'Captcha9kw'), (u'hooks', u'CaptchaBrotherhood'), (u'hooks', u'CaptchaTrader'), (u'hooks', u'ClickAndLoad'), (u'hooks', u'DebridItaliaCom'), (u'hooks', u'DebridItaliaComHook'), (u'hooks', u'EasybytezCom'), (u'hooks', u'EasybytezComHook'), (u'hooks', u'Ev0InFetcher'), (u'hooks', u'FastixRu'), (u'hooks', u'FastixRuHook'), (u'hooks', u'FreeWayMe'), (u'hooks', u'FreeWayMeHook'), (u'hooks', u'HighWayMe'), (u'hooks', u'HighWayMeHook'), (u'hooks', u'IRCInterface'), (u'hooks', u'LinkdecrypterCom'), (u'hooks', u'LinksnappyCom'), (u'hooks', u'LinksnappyComHook'), (u'hooks', u'MegaDebrid'), (u'hooks', u'MegaDebridEu'), (u'hooks', u'MegaDebridEuHook'), (u'hooks', u'MegaRapidoNet'), (u'hooks', u'MegaRapidoNetHook'), (u'hooks', u'MultiDebridCom'), (u'hooks', u'MultihostersCom'), (u'hooks', u'MultihostersComHook'), (u'hooks', u'MultishareCz'), (u'hooks', u'MultishareCzHook'), (u'hooks', u'MyfastfileCom'), (u'hooks', u'MyfastfileComHook'), (u'hooks', u'NoPremiumPl'), (u'hooks', u'NoPremiumPlHook'), (u'hooks', u'OverLoadMe'), (u'hooks', u'OverLoadMeHook'), (u'hooks', u'Premium4Me'), (u'hooks', u'PremiumTo'), (u'hooks', u'PremiumToHook'), (u'hooks', u'PremiumizeMe'), (u'hooks', u'PremiumizeMeHook'), (u'hooks', u'PutdriveCom'), (u'hooks', u'PutdriveComHook'), (u'hooks', u'RPNetBiz'), (u'hooks', u'RPNetBizHook'), (u'hooks', u'RapideoPl'), (u'hooks', u'RapideoPlHook'), (u'hooks', u'RealdebridCom'), (u'hooks', u'RealdebridComHook'), (u'hooks', u'RehostTo'), (u'hooks', u'RehostToHook'), (u'hooks', u'ReloadCc'), (u'hooks', u'RestartSlow'), (u'hooks', u'SimplyPremiumCom'), (u'hooks', u'SimplyPremiumComHook'), (u'hooks', u'SimplydebridCom'), (u'hooks', u'SimplydebridComHook'), (u'hooks', u'SmoozedCom'), (u'hooks', u'SmoozedComHook'), (u'hooks', u'UnrestrictLi'), (u'hooks', u'UnrestrictLiHook'), (u'hooks', u'Vipleech4uCom'), (u'hooks', u'WindowsPhoneToastNotify'), (u'hooks', u'XFileSharingPro'), (u'hooks', u'XMPPInterface'), (u'hooks', u'ZeveraCom'), (u'hooks', u'ZeveraComHook'), (u'hoster', u'File4safeCom'), (u'hoster', u'FileParadoxIn'), (u'hoster', u'Keep2shareCC'), (u'hoster', u'Keep2shareCc'), (u'hoster', u'Libgen'), (u'hoster', u'MegaDebrid'), (u'hoster', u'MegaNz'), (u'hoster', u'MultiDebridCom'), (u'hoster', u'NowDownloadEu'), (u'hoster', u'NowVideoAt'), (u'hoster', u'PandaPlaNet'), (u'hoster', u'PandaPlanet'), (u'hoster', u'Premium4Me'), (u'hoster', u'PutlockerCom'), (u'hoster', u'RapidshareCom'), (u'hoster', u'RealdebridComTorrent'), (u'hoster', u'ReloadCc'), (u'hoster', u'SendmywayCom'), (u'hoster', u'Share4webCom'), (u'hoster', u'ShareRapidCom'), (u'hoster', u'Solidfiles'), (u'hoster', u'UploadableCh'), (u'hoster', u'Vipleech4uCom'), (u'hoster', u'XFileSharingPro'), (u'internal', u'AbstractExtractor'), (u'internal', u'AdYouLike'), (u'internal', u'AdsCaptcha'), (u'internal', u'Hook'), (u'internal', u'MultiHook'), (u'internal', u'ReCaptcha'), (u'internal', u'SimpleDereferer'), (u'internal', u'SolveMedia'), (u'internal', u'XFSPAccount'), (u'internal', u'XFSPCrypter'), (u'internal', u'XFSPHoster'), (u'internal', u'utils')]
100     19.11.2019 14:12:37     INFO    ADDON UpdateManager: All plugins are up to date!
101     19.11.2019 14:13:08     INFO    ACCOUNT RealdebridCom: Adding user `asm*******`...
102     19.11.2019 14:13:08     DEBUG   ACCOUNT RealdebridCom: Reached login timeout for user `asm*******`
103     19.11.2019 14:13:08     INFO    ACCOUNT RealdebridCom: Login user `asm*******`...
104     19.11.2019 14:13:08     DEBUG   ACCOUNT RealdebridCom: LOAD URL https://api.real-debrid.com/rest/1.0/user | redirect=True | cookies=True | get={'auth_token': '**********'} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
105     19.11.2019 14:13:38     ERROR   ACCOUNT RealdebridCom: Could not login user `asm*******` | (28, 'Connection timed out after 30000 milliseconds')
106     19.11.2019 14:13:39     ERROR   ACCOUNT RealdebridCom: Could not load hoster list - invalid account, retry in 5 minutes
```
pyload restart with account already added:
```
108	19.11.2019 14:16:38	INFO	Starting pyLoad 0.4.9
109	19.11.2019 14:16:38	INFO	Using home directory: /home/pyload/.pyload
110	19.11.2019 14:16:38	DEBUG	Deleting old pidfile pyload.pid
111	19.11.2019 14:16:38	DEBUG	Remote activated: True
112	19.11.2019 14:16:45	DEBUG	created index of plugins
113	19.11.2019 14:16:45	INFO	Activated plugins: ClickNLoad, EventMapper, ExternalScripts, UnSkipOnFail, UpdateManager, UserAgentSwitcher, XFileSharing
114	19.11.2019 14:16:45	INFO	Deactivate plugins: AndroidPhoneNotify, AntiStandby, AntiVirus, BypassCaptcha, Captcha9Kw, Checksum, CloudFlareDdos, DeathByCaptcha, DeleteFinished, DiscordNotifier, DownloadScheduler, ExpertDecoders, ExtractArchive, HotFolder, IRC, ImageTyperz, JustPremium, LinkFilter, LinkdecrypterComHook, LogMarker, MergeFiles, MultiHome, PushBullet, PushOver, RestartFailed, SkipRev, TransmissionRPC, WindowsPhoneNotify, XMPP
115	19.11.2019 14:16:45	INFO	Downloadtime: True
116	19.11.2019 14:16:45	INFO	Starting ThriftBackend: 0.0.0.0:7227
117	19.11.2019 14:16:48	INFO	Starting builtin webserver: 0.0.0.0:8001
118	19.11.2019 14:16:48	INFO	Free space: 604.37 GiB
119	19.11.2019 14:16:49	INFO	Activating Accounts...
120	19.11.2019 14:16:49	INFO	ACCOUNT RealdebridCom: Adding user `asm*******`...
121	19.11.2019 14:16:49	DEBUG	ACCOUNT RealdebridCom: Reached login timeout for user `asm*******`
122	19.11.2019 14:16:49	INFO	ACCOUNT RealdebridCom: Login user `asm*******`...
123	19.11.2019 14:16:49	DEBUG	ACCOUNT RealdebridCom: LOAD URL https://api.real-debrid.com/rest/1.0/user | redirect=True | cookies=True | get={'auth_token': '**********'} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
124	19.11.2019 14:17:19	ERROR	ACCOUNT RealdebridCom: Could not login user `asm*******` | (28, 'Connection timed out after 30001 milliseconds')
125	19.11.2019 14:17:19	INFO	Activating Plugins...
126	19.11.2019 14:17:19	DEBUG	ADDON ExternalScripts: No script found under folder `pyload_start`
127	19.11.2019 14:17:19	DEBUG	ADDON UpdateManager: LOAD URL https://raw.githubusercontent.com/pyload/updates/master/plugins.txt | redirect=True | cookies=True | get={'v': '0.4.9'} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
128	19.11.2019 14:17:20	INFO	ADDON XFileSharing: Auto-discover new hosters
129	19.11.2019 14:17:20	DEBUG	ADDON XFileSharing: Pattern for hosters: (?:https?://(?:www\.)?)(?!(?:www\.)?(?:180upload\.com|clicknupload\.org|cramit\.in|ddl\.to|easybytez\.com|exashare\.com|filejoker\.net|fileom\.com|filerio\.com|fistfast\.net|hugefiles\.net|junocloud\.me|katfile\.com|movreel\.com|nosupload\.com|novafile\.com|rapidfileshare\.net|rarefile\.net|safesharing\.eu|secureupload\.eu|streamcloud\.eu|suprafiles\.me|tusfiles\.net|uploadrocket\.net|vidplay\.net))(?P(?:[\d.]+|[\w\-^_]{3,63}(?:\.[a-zA-Z]{2,})+)(?:\:\d+)?)/(?:embed-)?\w{12}(?:\W|$)
130	19.11.2019 14:17:20	INFO	ADDON XFileSharing: Auto-discover new crypters
131	19.11.2019 14:17:20	DEBUG	ADDON XFileSharing: Pattern for crypters: (?:https?://(?:www\.)?)(?!(?:www\.)?(?:easybytez\.com|tusfiles\.net))(?P(?:[\d.]+|[\w\-^_]{3,63}(?:\.[a-zA-Z]{2,})+)(?:\:\d+)?)/(?:user|folder)s?/\w+
132	19.11.2019 14:17:20	INFO	pyLoad is up and running
133	19.11.2019 14:17:22	ERROR	ACCOUNT RealdebridCom: Could not load hoster list - invalid account, retry in 5 minutes
134	19.11.2019 14:17:24	INFO	ADDON ClickNLoad: Proxy listening on 0.0.0.0:9666
135	19.11.2019 14:17:24	ERROR	ADDON ClickNLoad: [Errno 98] Address already in use
```

**Comment by GammaC0de** [comment]
Can you try installing pyload on windows and test the account there?

**Comment by asmoday83** [comment]
sorry für taking so long: I just installed it on a windows system and it works... looks like it's an issue with my system. thanks for your help

**Comment by GammaC0de** [comment]
👍 

Closing then.

**Comment by lollilol** [comment]
Hey, i have just have the exact same problem as you. I tried many things (including the username+password script generator thing, that i got from the logs) and it just wouldnt me let connect.


```
97 | 01.05.2020 15:15:04 | INFO | ACCOUNT RealdebridCom: Adding user `GG5*******`...
-- | -- | -- | --
98 | 01.05.2020 15:15:04 | INFO | ACCOUNT RealdebridCom: Login user `GG5*******`...
99 | 01.05.2020 15:15:34 | ERROR | ACCOUNT RealdebridCom: Could not login user `GG5*******` \| (28, 'Connection timed out after 30001 milliseconds')
100 | 01.05.2020 15:15:38 | ERROR | ACCOUNT RealdebridCom: Could not load hoster list - invalid account, retry in 5 minutes
101 | 01.05.2020 15:19:39 | INFO | ACCOUNT RealdebridCom: Updating account info for user `GG5*******`...
102 | 01.05.2020 15:19:39 | INFO | ACCOUNT RealdebridCom: Relogin user `GG5*******`...
103 | 01.05.2020 15:20:09 | ERROR | ACCOUNT RealdebridCom: Could not login user `GG5*******` \| (28, 'Connection timed out after 30000 milliseconds')
104 | 01.05.2020 15:20:09 | INFO | ACCOUNT RealdebridCom: Updating account info for user `GG5*******`...
105 | 01.05.2020 15:20:09 | INFO | ACCOUNT RealdebridCom: Relogin user `GG5*******`...
106 | 01.05.2020 15:20:12 | ERROR | ACCOUNT RealdebridCom: Could not load hoster list - invalid account, retry in 5 minutes
107 | 01.05.2020 15:20:39 | ERROR | ACCOUNT RealdebridCom: Could not login user `GG5*******` \| (28, 'Connection timed out after 30001 milliseconds')
108 | 01.05.2020 15:20:42 | ERROR | ACCOUNT RealdebridCom: Could not load hoster list - invalid account, deactivating
109 | 01.05.2020 15:20:42 | INFO | ACCOUNT RealdebridCom: Reverting back to default hosters
```

i then did
`curl 'https://api.real-debrid.com/rest/1.0/user?auth_token=*myauthtoken*` and it was successfull. My ip adress isnt blocked by them. It _should_ work, but it doesnt. It times out.

**Comment by GammaC0de** [comment]
@lollilol If you have a problem, please create another issue.

**Comment by OptimusGREEN** [comment]
For anyone else who stumbles upon this issue.
Download [GetRealDebridToken.py](https://github.com/pyload/pyload/files/4406037/GetRealdebridToken.zip) and run it and it will give you a username and password to login with via creating an app token.
